android/src/main/java/io/ltbl/bdkrn/BdkRnModule.kt

```diff
@@ -393,8 +394,12 @@ class BdkRnModule(reactContext: ReactApplicationContext) :
     @ReactMethod
     fun sync(id: String, blockChainId: String, result: Promise) {
         try {
-            getWalletById(id).sync(getBlockchainById(blockChainId), BdkProgress)
-            result.resolve(true)
+            Thread {
+                runOnUiThread {
+                    getWalletById(id).sync(getBlockchainById(blockChainId), BdkProgress)
+                    result.resolve(true)
+                }
+            }.start()
```

Comment:
Found that all is needed is not to run it on the `UiThread`

```suggestion
            Thread {
                getWalletById(id).sync(getBlockchainById(blockChainId), BdkProgress)
                result.resolve(true)
            }.start()
```

Reply:
Yes, reached a similar conclusion. Does this resolve the UI thread issue for you?

Reply:
Yes!